actor_blockstore/src/tracking.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+//#![cfg(feature = "tracking")]
+
+use std::cell::RefCell;
+
+use anyhow::Result;
+use cid::multihash::{self, Code};
+use cid::Cid;
+use fvm_ipld_blockstore::{Block, Blockstore};
+
+/// Stats for a [TrackingBlockstore] this indicates the amount of read and written data
+/// to the wrapped store.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub struct BSStats {
+    /// Number of reads
+    pub r: usize,
+    /// Number of writes
+    pub w: usize,
+    /// Bytes Read
+    pub br: usize,
+    /// Bytes Written
+    pub bw: usize,
+}
+
+/// Wrapper around `Blockstore` to tracking reads and writes for verification.
+/// This struct should only be used for testing.
+#[derive(Debug)]
+pub struct TrackingBlockstore<BS> {
```

Comment:
I'm not sure what the EVM actor tests use this for, but let's hold off copying it if that's the only consumer. And I can't see any use of the stats. Unless @Stebalien already has things to change about it here. 

I remember this was part of my initial desire for these implementations, but I've forgotten exactly what I wanted to add/change. Probably we want to remove the type parameter on MockRuntime and use a concrete blockstore specialised for testing, which will be in-memory and with tracking. But defer that to a follow-up. Ditto for the TestVM, removing the generic, and adding a store which can calculate actual gas usage of storage too.

Reply:
ping @Stebalien

Reply:
We used it in the EVM to tune the EVM's state-tree. It's useful for:

1. Tuning the "bitwidth" of AMTs/HAMTs.
2. Testing out new AMT/HAMT features.
3. Detecting regressions in state load/store patterns. This was our primary use-case after initial tuning.